상품 다중 조회 API 앤드포인트 개발

tableMinPark opened the pull request:
## 개요
- related issue : #40 

## 구현 내용
- 페이지, 사이즈로 다중 상품 조회
- 페아지, 사이즈가 없으면 400 으로 응답
- 페이지는 0부터 시작 (첫 페이지 번호가 0)
- 사이즈는 1 이상

tableMinPark committed:
feat : 상품 다중 조회 API 앤드포인트 개발

- 페이지, 사이즈로 다중 상품 조회
- 페이지 0 미만, 사이즈 1 미만인 경우에 400 으로 응답
- 등록된 상품 목록 반환 dto ReadAllItemResponse 생성
- Validated 를 통해 page, size 값 검증

tableMinPark committed:
feat : ItemController#readAllItem 메소드 명 수정

- readAllItem 을 readAllItems 로 수정

tableMinPark committed:
feat : ItemController#readAllItems @Validated 예외 발생 처리

- @RequestParam 의 @Validated 유효성 검사에서 실패하면 ConstraintViolationException 이 발생하는데, @ReuqestBody 처럼 ExceptionHandler 가 없어 예외가 발생하고 500 응답
- ConstraintViolationException 을 Advice 로 처리해서 400 으로 응답

tableMinPark committed:
test : readAllItems 테스트 작성

- 페이와 사이즈로 상품 목록을 조회하고 상품 리스트로 응답
- 페이지가 없을 시 400 응답
- 사이즈가 없을 시 400 응답
- 페이지와 사이즈 없이 조회하면 400 응답
- 페이지가 0 미만으로 조회하면 400과 실패 메세지 응답
- 사이즈가 1 미만으로 조회하면 400과 실패 메시지 응답

daengdaengLee commented on src/main/java/hello/until/item/controller/ItemController.java:
```diff
@@ -23,6 +27,12 @@ public ReadItemResponse readItem(@PathVariable long id) {
     }
 
+    @GetMapping
+    public ReadAllItemResponse readAllItems(@RequestParam @Min(value = 0, message = "페이지는 0 이상이여야 합니다.") Integer page,
+                                           @RequestParam @Min(value = 1, message = "사이즈는 1 이상이여야 합니다.") Integer size) {
```
page, size 는 required false 로 하고 default value 를 넣는 건 어떨까요?
그냥 /items 로 해도 기본값으로 첫 페이지 읽히게요.

tableMinPark replied:
그 부분까지 생각을 못했네요. 수정하겠습니다. :)

daengdaengLee commented on src/main/java/hello/until/item/controller/ItemController.java:
```diff
@@ -2,14 +2,18 @@
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.server.ResponseStatusException;
 
+@Validated
```
오옹.. 여기에 붙이면 어떤 식으로 동작하나요?

tableMinPark replied:
@ReuqestParam 입력 값 검증하기 위해서는 필요해서 붙였습니다!

@RequestBody에 쓰듯이 @RequestParam에 적용하니까 @Validated 가 적용이 안되더라구요!
찾아보니까 @RequestParam에 @Validated 적용하려면 Controller에 붙혀줘야 되서 붙였습니다!

daengdaengLee replied:
좋은 정보 감사합니다!

daengdaengLee reviewed (dismissed):
LGTM!
코멘트 한 번만 확인해주세요!

tableMinPark committed:
feat : ItemController#readAllItems 수정

- page, size의 값이 입력되지 않아도 기본값으로 조회가 가능하도록 수정
- page defaultValue 0 으로 지정
- size defaultValue 10 으로 지정

tableMinPark committed:
test : ItemControllerTest.ReadAll#readAllItemNoPageAndSize 수정

- page, size 값이 입력되지 않았을 때 page의 기본값 0, size의 기본값 10 으로 목록을 조회하는 테스트로 수정

daengdaengLee reviewed (dismissed):
LGTM!

tableMinPark committed:
Merge branch 'develop' into feat/api/readAll-item

daengdaengLee reviewed (approved):
LGTM
